contracts/src/v0.8/ccip/pools/GHO/UpgradeableBurnMintTokenPool.sol

```diff
@@ -0,0 +1,78 @@
+/// increases facilitator level.
+/// - Add GHO-Specific onlyOwner `directBurn` which burns liquidity & reduces facilitator level.
+/// - Remove i_token decimal check in UpgradeableTokenPool constructor
+
+/// @dev Pool whitelisting mode is set in the constructor and cannot be modified later.
+/// It either accepts any address as originalSender, or only accepts whitelisted originalSender.
+/// The only way to change whitelisting mode is to deploy a new pool.
+/// If that is expected, please make sure the token's burner/minter roles are adjustable.
+/// @dev This contract is a variant of BurnMintTokenPool that uses `burn(amount)`.
+contract UpgradeableBurnMintTokenPool is Initializable, UpgradeableBurnMintTokenPoolAbstract, ITypeAndVersion {
+  string public constant override typeAndVersion = "BurnMintTokenPool 1.5.1";
+
+  constructor(
+    address token,
+    uint8 localTokenDecimals,
+    address rmnProxy,
+    bool allowListEnabled
+  ) UpgradeableTokenPool(IBurnMintERC20(token), localTokenDecimals, rmnProxy, allowListEnabled) {}
+
+  function initialize(address owner_, address[] memory allowlist, address router) external initializer {
+    if (router == address(0) || owner_ == address(0)) revert ZeroAddressNotAllowed();
+
+    _transferOwnership(owner_);
+    s_router = IRouter(router);
+    if (i_allowlistEnabled) _applyAllowListUpdates(new address[](0), allowlist);
+  }
+
+  /// @notice This function allows the owner to mint `amount` tokens on behalf of the pool
+  /// and transfer them to `to`. This is GHO-Specific and is called to match the facilitator
+  /// level of the new pool with the old pool such that it can burn the bridged supply once
+  /// the old pool is deprecated. The old pool is then expected to burn `amount` of tokens
+  /// so that it can be removed as a facilitator on GHO (ideally using `directBurn`).
+  /// @dev This is only called while offboarding an old token pool (or facilitator) in favor
+  /// of this pool.
+  /// @param to The address to which the minted tokens will be transferred. This needs to
+  /// be the old token pool,
+  /// or the facilitator being offboarded.
+  /// @param amount The amount of tokens to mint and transfer to old pool.
+  function directMint(address to, uint256 amount) external onlyOwner {
+    IBurnMintERC20(address(i_token)).mint(to, amount);
+  }
+
+  /// @notice This function allows the owner to burn `amount` of the pool's token. This is
+  /// expected to be called while migrating facilitators by offboarding this facilitator in
+  /// favor of a new token pool.
+  /// @dev New token pool should mint and transfer liquidity to this pool (since this pool
+  /// does not hold tokens at any point in time) which can be burnt and hence will reduce
+  /// the facilitator bucket level on GHO. The naming convention is inspired from  that in
+  /// LockRelease type token pools for the sake of consistency.
+  /// @param amount The amount of tokens to burn.
```

Comment:
same
```suggestion
  /// @notice Burn an amount of tokens with no additional logic. 
  /// @dev This GHO-specific functionality is designed for migrating bucket levels between facilitators. The old 
  /// pool is expected to burn `amount` of tokens, allowing this pool to mint an equivalent `amount`.
  /// @param amount The amount of tokens to burn
```

Reply:
again, we mint on new pool before we burn on old pool, so maybe adjust wording similar to above, wdyt

Reply:
yes, works

Reply:
[4e0e90f](https://github.com/aave/ccip/pull/20/commits/4e0e90fbded84e592affeea4dacf5e597f6a251b)